pkg/sdkserver/localsdk.go

```diff
@@ -58,24 +58,64 @@ var (
 // is being run for local development, and doesn't connect to the
 // Kubernetes cluster
 type LocalSDKServer struct {
-	gsMutex          sync.RWMutex
-	gs               *sdk.GameServer
-	update           chan struct{}
-	updateObservers  sync.Map
-	requestSequence  []string
-	expectedSequence []string
-	testMode         bool
+	gsMutex               sync.RWMutex
+	gs                    *sdk.GameServer
+	update                chan struct{}
+	updateObservers       sync.Map
+	requestSequence       []string
+	expectedSequence      []string
+	gsState               agonesv1.GameServerState
+	gsReserveDuration     *time.Duration
+	reserveTimer          *time.Timer
+	testMode              bool
+	validStateTransitions map[string]bool
+}
+
+func insertStateTransition(m map[string]bool, from, to agonesv1.GameServerState) {
+	m[string(from)+string(to)] = true
+}
+
+func initValidStateTransitions() map[string]bool {
```

Comment:
I'm wondering if we really care about this?

What are we trying to prevent here? I'm just wondering if this is going put some limitations on what's possible with the local test server, and add extra maintenance down the road for us.

What's the worst that happens if we don't validate state transitions locally?

Reply:
Main intention was to prevent developers from using wrong assumptions while debugging their game servers locally.
Additionally this would be a place in go code where we state all possible edges of state diagram.
I imagine that we can add a flag to validate this transitions or not so that it would be optional.

Reply:
I'm just trying to think of what could be a bad transition?

Moving from Shutdown -> Ready/Allocated/Reserved ?

Even then - I'm not sure if we should stop people. If the developer wants to go in that loop, they would have to restart the gameserver and sdk every time, and that could slow them down. Fast development iteration loops are a very good thing, so should we be putting extra friction in people's way?

If people really want to move from Shutdown -> Ready they can change the backing file anyway, and it (should) overwrite what's in the local sdk. So I'm not sure this is necessary?

Reply:
Yes, that's right. This is redundant for local development and those steps are well documented in GS State transitions schema.